File: src/main/java/com/tastyload/tastyload/placeGroup/entity/PlaceGroupItemEntity.java
package com.tastyload.tastyload.placeGroup.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

import com.tastyload.tastyload.place.PlaceEntity;

@Entity
@Table(name = "place_group_item")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class PlaceGroupItemEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "place_group_id", referencedColumnName = "id")
    private PlaceGroupEntity placeGroup;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "place_id", referencedColumnName = "id")
    private PlaceEntity place;

    @Column(name = "created_at")
    private LocalDateTime createdAt;
    
}

File: backend/src/place-groups/entities/place-group-item.entity.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
  UpdateDateColumn,
  ManyToOne,
  JoinColumn,
} from 'typeorm';
import { PlaceGroup } from './place-group.entity';
import { Place } from '../../places/entities/place.entity';

@Entity('place_group_items')
export class PlaceGroupItem {
  @PrimaryGeneratedColumn()
  id: number;

  @ManyToOne(() => PlaceGroup, (placeGroup) => placeGroup.items, { lazy: true })
  @JoinColumn({ name: 'place_group_id', referencedColumnName: 'id' })
  placeGroup: PlaceGroup;

  @ManyToOne(() => Place, { lazy: true })
  @JoinColumn({ name: 'place_id', referencedColumnName: 'id' })
  place: Place;

  @Column({ nullable: true })
  note: string;

  @CreateDateColumn({ name: 'created_at' })
  createdAt: Date;

  @UpdateDateColumn({ name: 'updated_at' })
  updatedAt: Date;
}